src/Objects/_PVector.js

```diff
@@ -0,0 +1,283 @@
+          cy = v1.z*v2.x - v2.z*v1.x,
+          cz = v1.x*v2.y - v2.x*v1.y;
+    return target? target.set(cx, cy, cz) : new _PVector(cx, cy, cz);
+  };
+
+  _PVector.angleBetween = function(v1, v2) {
+    if (!v1.x && !v1.y && !v1.z || !v2.x && !v2.y && !v2.z) { return 0; }
+    const amt = _PVector.dot(v1, v2) / Math.sqrt(v1.magSq() * v2.magSq());
+    return amt <= -1? Math.PI : amt >= 1? 0 : Math.acos(amt);
+  };
+
+  _PVector.lerp = function(v1, v2, amt) {
+    return v1.copy().lerp(v2, amt);
+  };
+
+  _PVector.add = function(v1, v2, target) {
+    return target? target.set(v1.x + v2.x, v1.y + v2.y, v1.z + v2.z)
+                 : new _PVector(v1.x + v2.x, v1.y + v2.y, v1.z + v2.z);
+  };
+
+  _PVector.sub = function(v1, v2, target) {
+    return target? target.set(v1.x - v2.x, v1.y - v2.y, v1.z - v2.z)
+                 : new _PVector(v1.x - v2.x, v1.y - v2.y, v1.z - v2.z);
+  };
+
+  _PVector.mult = function(v, n, target) {
+    if (typeof n === 'number') {
+      return target? target.set(v.x*n, v.y*n, v.z*n)
+                   : new _PVector(v.x*n, v.y*n, v.z*n);
+    }
+    return target? target.set(v.x*n.x, v.y*n.y, v.z*n.z)
+                 : new _PVector(v.x*n.x, v.y*n.y, v.z*n.z);
+  };
+
+  _PVector.div = function(v, n, target) {
+    if (typeof n === 'number') {
+      return target? target.set(v.x/n, v.y/n, v.z/n)
+                   : new _PVector(v.x/n, v.y/n, v.z/n);
+    }
+    return target? target.set(v.x/n.x, v.y/n.y, v.z/n.z)
+                 : new _PVector(v.x/n.x, v.y/n.y, v.z/n.z);
+  };
+
+  _PVector.prototype = {
+    copy: function() {
+      return new _PVector(this.x, this.y, this.z);
+    },
+    get: function(target) {
+      if (!arguments.length) { return this.copy(); } // @Deprecated
+      if (typeof target !== 'object') { return this.array(); }
+      target[0] = this.x;
+      target[1] = this.y;
+      target[2] = this.z;
+      return target;
+    },
+    set: function(v, y, z) {
+      const len = arguments.length;
+      if (len > 1) {
+        this.x = v;
+        this.y = y;
+        if (len > 2) { this.z = z; }
+      } else if (len === 1) {
+        this.set(v.x || v[0] || 0,
+                 v.y || v[1] || 0,
+                 v.z || v[2] || 0);
+      } else { argsErr('set', len, 1); }
+      return this;
+    },
+    array: function() {
+      return [this.x, this.y, this.z];
+    },
+    normalize: function(target) {
+      const m = this.mag(),
+            canDivide = m !== 0 && m !== 1;
+      if (!arguments.length) { return canDivide? this.div(m) : this; }
+      return canDivide? _PVector.div(this, m, target)
+                      : target? target.set(this) : this.copy();
+    },
+    limit: function(max) {
+      return this.magSq() > max*max? this.normalize().mult(max) : this;
+    },
+    heading: function() {
+      //return -Math.atan2(-this.y, this.x);
+      return Math.atan2(this.y, this.x);
+    },
+    heading2D: null, // @Deprecated
```

Comment:
> Also note that null is a value to avoid in JavaScript,

That is exactly the use case for `null`. That is, when we create a property but it still got nothing in it,  we assign `null`.
`undefined` in turn conveys that something isn't even exist!